rust/operator-binary/src/spark_k8s_controller.rs

```diff
@@ -767,6 +778,17 @@ fn submit_job_config_map(
     );
 
     if let Some(product_config) = product_config {
+        cm_builder.add_data(
```

Comment:
We have this bit of code three times in the controllers: is it worth pushing it down into to_spark_env_sh_string?

Reply:
(the `.get()...iter()` bit, that is)

Reply:
I totally agree, that this verbose and repeated code. The `to_spark_env_sh_string` function is modelled after the function `to_java_properties_string`. Both have the input parameter type `Iterator<Item = (&'a String, &'a String)>` which is very generic and (in a Rust perspective) clean. The call on the other hand is an unclean `.get(...).cloned().unwrap_or_default().iter()` construct. If we make the call clean like `to_spark_env_sh_string(product_config, SPARK_ENV_SH_FILE_NAME)` then the function becomes more unclean. Intermediate solutions like `to_spark_env_sh_string(product_config.get(&PropertyNameKind::File(SPARK_ENV_SH_FILE_NAME.to_string())))` would also be possible and semi-unclean on both sides.

In my opinion this is just the usual (unavoidable) Rust rat tail. But I am open for a proposal and willing to implement it.

Reply:
I'd say we leave it then, if there is no solution that makes things obviously better.